src/adapter/src/coord/statement_logging.rs

```diff
@@ -719,10 +719,16 @@ impl Coordinator {
         }
         let (ps_record, ps_uuid) = self.log_prepared_statement(session, logging)?;
 
-        let ev_id = Uuid::new_v4();
+        let uuid = match session.qcell_rw(logging) {
+            PreparedStatementLoggingInfo::AlreadyLogged { uuid } => *uuid,
+            PreparedStatementLoggingInfo::StillToLog { prepared_at, .. } => {
+                epoch_to_uuid_v7(prepared_at)
+            }
+        };
+
         let now = self.now();
         self.record_statement_lifecycle_event(
-            &StatementLoggingId(ev_id),
+            &StatementLoggingId(uuid),
```

Comment:
Do I understand correctly that this has changed the behavior of statement lifecycle logging around prepared statements? Before this, we used to assign a new id for each execution of a prepared statement. After the change, we use the same id for all the executions of a prepared statement. This is because the above `log_prepared_statement` call will set `logging` to `PreparedStatementLoggingInfo::AlreadyLogged` the first time the prepared statement is executed (unless throttled), and then the new code will just use that `AlreadyLogged` id.

Was this intended?

Reply:
that was not intended. that's an excellent point. I believe we should instead be using the current time generate a correctly ordered log?

Reply:
I will fix with: https://github.com/MaterializeInc/database-issues/issues/9242

Reply:
thanks so much for finding this bug!

Reply:
fix: https://github.com/MaterializeInc/materialize/pull/32378

Reply:
Thanks! Let's discuss on that PR.